### PR TITLE
[homekit] add support for QR code based pairing

### PR DESCRIPTION
**Pull request by yfre**
replace #9458 that was wrongly pushed with too many files changed. sorry for that. 
----

small goodie for OH3 - support for QR code based pairing to HomeKit.

this PR includes:

update of java-hap library to the next version. this version add support for qr code, linked services and new jmDNS version
add QR code to configuration
make use of ConfigurationAdmin in order to update Homekit configuration
update README: pairing process using QR code and adding metadata using new UI

### CODE REVIEW

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java
Comment (J-N-K) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java:65]:
```suggestion
    private final NetworkAddressService networkAddressService;
    private final ConfigurationAdmin configAdmin;
```

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java
Comment (J-N-K) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java:72]:
```suggestion
    private final HomekitChangeListener changeListener;
```

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java
Comment (J-N-K) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java:68]:
```suggestion
    private HomekitAuthInfoImpl authInfo;
    private HomekitSettings settings;
```

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java
Comment (J-N-K) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java:90]:
```suggestion
            logger.warn("Cannot activate HomeKit binding. {}", e.getMessage());
            throw e;
```

If the constructor throws an exception, the framework should try to re-initialize.

**Review by J-N-K — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java
Comment (yfre) [reply bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java:68]: thx, good one. 
i have not thought about @NonNullByDefault({}) after changed the injection to constructor injection

**Review by J-N-K — APPROVED** [review]

**Comment by yfre** [comment]
@J-N-K  the build fails probably due missing Java-HAP. how you did it last time? do i need to add bintray as repository?

**Comment by J-N-K** [comment]
@yfre Looks like your comment is missing. If you make a PR with all changes against my repo, I‘ll release 2.1.0.OH immediately.

**Comment by yfre** [comment]
> @yfre Looks like your comment is missing. If you make a PR with all changes against my repo, I‘ll release 2.1.0.OH immediately.

@J-N-K thank you! just created the PR. let me know if it needs to be changed

File: bundles/org.openhab.io.homekit/pom.xml
Comment (J-N-K) [inline bundles/org.openhab.io.homekit/pom.xml:26]:
```suggestion
      <version>2.1.0.OH</version>
```

File: bundles/org.openhab.io.homekit/pom.xml
Comment (J-N-K) [inline bundles/org.openhab.io.homekit/pom.xml:24]: revert that change

**Review by J-N-K — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.io.homekit/pom.xml
Comment (yfre) [reply bundles/org.openhab.io.homekit/pom.xml:26]: done

File: bundles/org.openhab.io.homekit/pom.xml
Comment (J-N-K) [inline bundles/org.openhab.io.homekit/pom.xml:24]:
```suggestion
      <groupId>com.github.j-n-k</groupId>
```

**Comment by yfre** [comment]
@J-N-K do we need a second review before we can merge it?

File: bundles/org.openhab.io.homekit/src/main/resources/OH-INF/config/config.xml
Comment (fwolter) [inline bundles/org.openhab.io.homekit/src/main/resources/OH-INF/config/config.xml:28]:
```suggestion
			<description>Scan QR code with home app to add openHAB as HomeKit bridge. </description>
```

File: bundles/org.openhab.io.homekit/src/main/resources/OH-INF/config/config.xml
Comment (fwolter) [inline bundles/org.openhab.io.homekit/src/main/resources/OH-INF/config/config.xml:27]: This context is not documented here https://www.openhab.org/docs/developer/bindings/config-xml.html#xml-structure-for-configuration-descriptions Are you sure it exists?

File: bundles/org.openhab.io.homekit/src/main/resources/OH-INF/config/config.xml
Comment (fwolter) [inline bundles/org.openhab.io.homekit/src/main/resources/OH-INF/config/config.xml:41]: Words in labels should be capitalized (except prepositions and so on). See https://www.openhab.org/docs/developer/bindings/thing-xml.html#formatting-labels-and-descriptions

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.io.homekit/src/main/resources/OH-INF/config/config.xml
Comment (yfre) [reply bundles/org.openhab.io.homekit/src/main/resources/OH-INF/config/config.xml]: thank you for the review. good points. fixed

**Review by fwolter — APPROVED** [review]

**Pull request merged**
